torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -776,6 +776,22 @@ def _full_post_state_dict_hook(
         state_dict: "OrderedDict[str, torch.Tensor]",
         prefix: str,
     ) -> "OrderedDict[str, torch.Tensor]":
+        """
+        Hook that runs after model.state_dict() is called before returning result to
+        user. For FSDP, we may have to clone the tensors in state_dict as params go
+        back to sharded version after summon_full_params ends, and also remove
+        "_fsdp_wrapped_module" prefix.
+        """
+        for key in state_dict.keys():
+            # Due to recursive call of summon_full_params, avoid unnecessary reclone of
+            if (
+                self.training_state == TrainingState_.SUMMON_FULL_PARAMS and
```

Comment:
Curious, is there any chance that this hook will be called with `self.training_state` is not `SUMMON_FULL_PARAMS`?

Reply:
Good point, it should never be called as far as I understand. I'll see if I can actually just change this to an assert.